server/src/test/java/org/elasticsearch/action/search/FetchSearchPhaseTests.java

```diff
@@ -749,4 +779,150 @@ private static void addProfiling(boolean profiled, QuerySearchResult queryResult
     private static ProfileResult fetchProfile(boolean profiled) {
         return profiled ? new ProfileResult("fetch", "fetch", Map.of(), Map.of(), FETCH_PROFILE_TIME, List.of()) : null;
     }
+
+    public void testFetchTimeoutWithPartialResults() throws IOException {
+        Directory dir = newDirectory();
+        RandomIndexWriter w = new RandomIndexWriter(random(), dir);
+        w.addDocument(new Document());
+        IndexReader r = w.getReader();
+        w.close();
+        try {
+            ContextIndexSearcher contextIndexSearcher = new ContextIndexSearcher(r, null, null, new QueryCachingPolicy() {
```

Comment:
NIT: maybe dry up the identical `ContextIndexSearcher` overrides :) ?

Reply:
done!